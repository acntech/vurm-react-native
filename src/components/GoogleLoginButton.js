import { GoogleSocialButton } from "react-native-social-buttons";
import * as Google from "expo-auth-session/providers/google";
import { GoogleAuthProvider, signInWithCredential } from "firebase/auth";
import { useEffect, useState } from "react";
import { auth } from "../firebase/auth";

const GoogleLoginButton = ({
  buttonViewStyle,
  setUser,
  setDisabled,
  disabled,
}) => {
  const [request, response, promptAsync] = Google.useIdTokenAuthRequest({
    clientId:
      "757230829578-3dq8g736piopuu5dn0hdo5slq38aeuu9.apps.googleusercontent.com",
  });
  const handlePress = () => {
    setDisabled(true);
    promptAsync().finally(() => {
      setDisabled(false);
    });
  };

  useEffect(() => {
    if (response?.type === "success") {
      // Build Firebase credential with the Google ID token.
      const { id_token } = response.params;

      const credential = GoogleAuthProvider.credential(id_token);
      signInWithCredential(auth, credential)
        .then(() => {
          setUser(auth.currentUser);
        })
        .catch((error) => {
          console.log(error);
        });
    }
  });
  return (
    <GoogleSocialButton
      buttonViewStyle={buttonViewStyle}
      onPress={handlePress}
      disabled={disabled}
    />
  );
};

export default GoogleLoginButton;
