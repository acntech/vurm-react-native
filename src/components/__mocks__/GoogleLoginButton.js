import { GoogleSocialButton } from "react-native-social-buttons";
import { auth } from "../../firebase/auth";
const GoogleLoginButton = ({
  buttonViewStyle,
  setUser,
  setDisabled,
  disabled,
}) => {
  const handlePress = () => {
    setUser({ uid: "testUid", displayName: "Test Person" });
  };
  return (
    <GoogleSocialButton
      buttonViewStyle={buttonViewStyle}
      onPress={handlePress}
      disabled={disabled}
    />
  );
};

export default GoogleLoginButton;
