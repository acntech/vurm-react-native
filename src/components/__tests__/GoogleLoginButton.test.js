import React from "react";
import { render } from "@testing-library/react-native";
import GoogleLoginButton from "../GoogleLoginButton";

jest.mock("expo-auth-session/providers/google", () => {
  const originalModule = jest.requireActual(
    "expo-auth-session/providers/google"
  );

  //Mock the default export and named export 'foo'
  return {
    __esModule: true,
    ...originalModule,
    useIdTokenAuthRequest: jest.fn(() => [undefined, undefined, undefined]),
  };
});

describe("<GoogleLoginButton />", () => {
  it("renders correctly", () => {
    // Arrange & Act
    const tree = render(<GoogleLoginButton />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });
});
