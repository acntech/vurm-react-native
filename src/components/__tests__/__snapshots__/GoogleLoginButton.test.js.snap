// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleLoginButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#fff",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "margin": 5,
      "opacity": 1,
      "width": 220,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 25,
        "marginLeft": 15,
        "padding": 10,
        "resizeMode": "stretch",
        "width": 25,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#575757",
        "marginLeft": 15,
        "marginRight": 20,
      }
    }
  >
    Sign in with Google
  </Text>
</View>
`;
